### ISSUE TITLE
Did the devit project  can work on the task of coco2014 few-shot object detection ?

### ISSUE THREAD

**Issue by qyu21490** [RuoyuChen10/CCL-FSOD#4]
I attempted to transplant your work to the few-shot object detection task of coco2014, but the accuracy dropped. I'm not sure what the reason is. The knowledge_martix_path I used is "KnowledgeMatrix/text_relationship/coco/coco-word-embedding.npy". Have you tried this task? Could you share some experience?

| $方法$ | $AP$ | $AP_b$ | $AP_n$ | $描述$ |
| ----- | ----- | ----- | ----- | ----- |
| DeViT default | 29.4 | 28.0 | 33.7 |
| CCL | 28.9 | 27.5 | 33.1 |

**Comment by RuoyuChen10** [comment]
> I attempted to transplant your work to the few-shot object detection task of coco2014, but the accuracy dropped. I'm not sure what the reason is. The knowledge_martix_path I used is "KnowledgeMatrix/text_relationship/coco/coco-word-embedding.npy". Have you tried this task? Could you share some experience?
> 
> 方
> 法
> 𝐴
> 𝑃
> 𝐴
> 𝑃
> 𝑏
> 𝐴
> 𝑃
> 𝑛
> 描
> 述
> DeViT default	29.4	28.0	33.7	
> CCL	28.9	27.5	33.1

Thank you for your attention to our method. Because the MS COCO dataset has a sufficient number of samples, some special training methods of few-shot learning may not have significant effects on MS COCO training. I think some potential improvement methods include 1) Try not to use extra tricks during pre-training, and use some methods such as this project during fine-tuning; 2) Pay attention to the training rounds, and try to save the model and evaluate it multiple times to select the best effect, because few-shot training is difficult to produce a stable model; 3) Use some methods to increase the weight of important scores in the knowledge matrix, such as adding a log function and then normalizing. I hope these can help you solve the problem. If you have other questions, please feel free to communicate.

**Comment by qyu21490** [comment]
Thanks. 
I conducted few-shot object detection training on COCO2014 (60 base cases, 20 novel cases). Should I choose "KnowledgeMatrix/text_relationship/coco/coco-word-embedding.npy" as the KnowledgeMatrix?

**Comment by RuoyuChen10** [comment]
> Thanks. I conducted few-shot object detection training on COCO2014 (60 base cases, 20 novel cases). Should I choose "KnowledgeMatrix/text_relationship/coco/coco-word-embedding.npy" as the KnowledgeMatrix?

I need to check this because the object detection methods based on the detectron2 framework, such as DeFRCN, coco2014, have the base classes and novel classes ordered alternately. DeVIT, on the other hand, relies on prototypes and has the base classes and novel classes ordered separately.

It's also important to note that our approach is fine-tuning-based. Similar approaches based on DeViT first train on the base class, then directly test the DeViT baseline on the prototype of the novel class. However, our approach follows DeViT base training and then performs fine-tuning simultaneously. Fine-tuning utilizes data from both the base and novel classes, and incorporates a knowledge matrix and other modules.

When fine-tuning COCO, I recommend reducing the learning rate and testing the accuracy every 50 to 100 steps. This may improve the performance of few-shot object detectors.

I hope this helps you verify your performance on the COCO 2014 dataset. If you have any further questions, please feel free to contact me.
